presto-geospatial/src/main/java/io/prestosql/plugin/geospatial/GeoFunctions.java

```diff
@@ -635,6 +638,93 @@ public static Double lineLocatePoint(@SqlType(GEOMETRY_TYPE_NAME) Slice lineSlic
         return new LengthIndexedLine(line).indexOf(point.getCoordinate()) / line.getLength();
     }
 
+    @SqlNullable
+    @Description("Returns a Point interpolated along a LineString at the fraction given.")
+    @ScalarFunction("line_interpolate_point")
+    @SqlType(GEOMETRY_TYPE_NAME)
+    public static Slice lineInterpolatePoint(
+            @SqlType(GEOMETRY_TYPE_NAME) Slice input,
+            @SqlType(StandardTypes.DOUBLE) double distanceFraction)
+    {
+        OGCGeometry geometry = deserialize(input);
+        if (geometry.isEmpty()) {
+            return null;
+        }
+
+        List<Point> interpolatedPoints = interpolatePoints(geometry, distanceFraction, false);
+        return serialize(createFromEsriGeometry(interpolatedPoints.get(0), null));
+    }
+
+    @SqlNullable
+    @Description("Returns an array of Points interpolated along a LineString.")
+    @ScalarFunction("line_interpolate_points")
```

Comment:
The postgis implementation is a little different in that this method will return a single point instead of a multipoint if the result contains only one; since this always returns an array, it might make more sense to not take a `repeated` param and default to `true` instead.